src/System.Text.Primitives/System/Text/Formatting/Internal/TimeFormatter.Utf16.cs

```diff
@@ -0,0 +1,322 @@
+        private const char Comma = ',';
+        private const char Minus = '-';
+        private const char Period = '.';
+        private const char Plus = '+';
+        private const char Slash = '/';
+        private const char Space = ' ';
+
+        private const char TimeMarker = 'T';
+        private const char UtcMarker = 'Z';
+
+        private const char GMT1 = 'G';
+        private const char GMT2 = 'M';
+        private const char GMT3 = 'T';
+
+        private static readonly char[][] DayAbbreviations = new char[][]
+        {
+                new char[] { 'S', 'u', 'n' },
+                new char[] { 'M', 'o', 'n' },
+                new char[] { 'T', 'u', 'e' },
+                new char[] { 'W', 'e', 'd' },
+                new char[] { 'T', 'h', 'u' },
+                new char[] { 'F', 'r', 'i' },
+                new char[] { 'S', 'a', 't' },
+        };
+
+        private static readonly char[][] MonthAbbreviations = new char[][]
+        {
+                new char[] { 'J', 'a', 'n' },
+                new char[] { 'F', 'e', 'b' },
+                new char[] { 'M', 'a', 'r' },
+                new char[] { 'A', 'p', 'r' },
+                new char[] { 'M', 'a', 'y' },
+                new char[] { 'J', 'u', 'n' },
+                new char[] { 'J', 'u', 'l' },
+                new char[] { 'A', 'u', 'g' },
+                new char[] { 'S', 'e', 'p' },
+                new char[] { 'O', 'c', 't' },
+                new char[] { 'N', 'o', 'v' },
+                new char[] { 'D', 'e', 'c' },
+        };
+
+        #endregion Constants
+
+        public static bool TryFormatG(DateTime value, TimeSpan offset, Span<byte> buffer, out int bytesWritten)
+        {
+            const int MinimumCharsNeeded = 19;
+
+            int charsNeeded = MinimumCharsNeeded;
+            if (offset != PrimitiveFormatter.NullOffset)
+            {
+                charsNeeded += 7; // Space['+'|'-']hh:ss
+            }
+
+            bytesWritten = charsNeeded * sizeof(char);
+            if (buffer.Length < bytesWritten)
+            {
+                bytesWritten = 0;
+                return false;
+            }
+
+            Span<char> dst = buffer.NonPortableCast<byte, char>();
+            ref char utf16Bytes = ref dst.DangerousGetPinnableReference();
+
+            FormattingHelpers.WriteDigits(value.Month, 2, ref utf16Bytes, 0);
+            Unsafe.Add(ref utf16Bytes, 2) = Slash;
+
+            FormattingHelpers.WriteDigits(value.Day, 2, ref utf16Bytes, 3);
+            Unsafe.Add(ref utf16Bytes, 5) = Slash;
+
+            FormattingHelpers.WriteDigits(value.Year, 4, ref utf16Bytes, 6);
+            Unsafe.Add(ref utf16Bytes, 10) = Space;
+
+            FormattingHelpers.WriteDigits(value.Hour, 2, ref utf16Bytes, 11);
+            Unsafe.Add(ref utf16Bytes, 13) = Colon;
+
+            FormattingHelpers.WriteDigits(value.Minute, 2, ref utf16Bytes, 14);
+            Unsafe.Add(ref utf16Bytes, 16) = Colon;
```

Comment:
Extra variables means more IL instructions. These are constants and therefore can be optimized.